### PR TITLE
replace sample Caddyfile with a number of examples

### PR DESCRIPTION
**Pull request by willnorris**
These examples don't demonstrate all of the various config options since most are pretty straightforward and not terribly interesting.  Instead, it focuses on the different plugins and types of use-cases they enable.

Also fix some inconsistent indentation in README config samples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,22 @@ Use [xcaddy](https://github.com/caddyserver/xcaddy) to build Caddy with the Tail
 xcaddy build v2.7.6 --with github.com/tailscale/caddy-tailscale
 ```
 
+### Running examples
+
+Multiple example configurations are provided in the [examples directory].
+These examples expect an [auth key] to be set in the `TS_AUTHKEY` environment variable.
+All nodes registered while running these examples will be ephemeral and removed after disconnect.
```
Comment (irbekrm) [inline README.md:28]: > will be ephemeral

I guess that depends on whether the auth key is ephemeral, right? Maybe reword like 'create [ephemeral auth key][link] for nodes to be removed after disconnect'?

File: README.md [hunk 2]
```diff
@@ -21,6 +21,22 @@ Use [xcaddy](https://github.com/caddyserver/xcaddy) to build Caddy with the Tail
...
+See the comments in the individual files for details.
+
+Run them with:
+
+```
+export TS_AUTHKEY=<tskey-auth-XXXXX>
+./caddy run -a caddyfile -c example/<file>
```
Comment (irbekrm) [inline README.md:35]: Nit:
```suggestion
./caddy run -a caddyfile -c examples/<file>
```

**Review by irbekrm — APPROVED** [review]
The examples are awesome and really well documented too- I found them super helpful!

File: README.md (hunk 2 above)
Comment (willnorris) [reply README.md]: done
